Megapass on Chapter 13 (#518)
browserengineering/book: book/animations.md

    Time in render on average: 23ms

That's about three times faster, almost fast enough to hit sixty
frames per second. (And on your computer, you'll likely see even more

> speedup than I did, so for you it might already be fast enough.) But

Author: Great point.

        # ...
```

So simple and elegant! Now, on every frame, we are simply splitting
the display list into composited layers and the draw display list, and
then running each of those it their own phase. We're now half-way

> toward getting super-smooth animations. What remains is *not*

Author: Rephrased to make clearer.

def style(node, rules, tab):
    if old_style:
        transitions = diff_styles(old_style, node.style)
        for property, (old_value, new_value, num_frames) \
            in transitions.items():
            if property in ANIMATED_PROPERTIES:
                tab.set_needs_render()
                AnimationClass = ANIMATED_PROPERTIES[property]
                animation = AnimationClass(
                    old_value, new_value, num_frames)
                node.animations[property] = animation
                node.style[property] = animation.animate()
```

Any time a property listed in a `transition` changes its value, we'll
create an animation and get ready to run it.^[Note that we need to
call `set_needs_render` here to make sure that the animation will run
on the next frame.]

With the animations created, we now need to run them on every later
frame. Basically, on every frame, we iterate through all the active
animations on the page and call `animate` on them. Since these

> animations cousin to JavaScript using `requestAnimationFrame`, let's

Author: Weird choice of word, rephrased.

[cssfilter]: https://developer.mozilla.org/en-US/docs/Web/CSS/filter

[^overlap-example]: It's not possible to create the overlapping

> squares example of this section without something way of moving an

Reviewer: ```suggestion
squares example of this section without some way of moving an
```

element around. Real browsers have many methods for this, such as
[position].

[position]: https://developer.mozilla.org/en-US/docs/Web/CSS/position

> The `transform` CSS property in quite general and lets you apply [any

Reviewer: ```suggestion
The `transform` CSS property is quite powerful, and lets you apply [any
```

                background-color:lightgreen;
                transform:translate(0px, 0px)"></div>

> Support these transform is simple. First let's parse the property

Reviewer: ```suggestion
Supporting these transforms is simple. First let's parse the property
```

Summary
=======

> This chapter introduce animations. The key takeaways you should

Reviewer: ```suggestion
This chapter introduces animations. The key takeaways you should
```
